src/utils/hooks/useProtectedPerson.ts

```
@@ -0,0 +1,18 @@
+/**
+ * React hook that retrieves protected person data based on a CristinPerson object or person ID.
+ * Uses `useFetchProtectedPerson` to fetch data unless an existing person object is provided.
+ * @param person - CristinPerson object or person ID string.
+ * @param searchEnabled - Optional flag to enable or disable fetching (default: true).
+ * @returns An object containing the resolved person data and the query result.
+ */
+export const useProtectedPerson = (person: CristinPerson | string, searchEnabled = true) => {
+  const personId = typeof existingPerson === 'string' ? existingPerson : existingPerson.id;
+  const existingPersonObject = typeof existingPerson === 'object' ? existingPerson : undefined;
+  const personQuery = useFetchProtectedPerson(personId, { enabled: searchEnabled && !existingPersonObject });
+  const person = existingPersonObject ?? personQuery.data;
```

Biome (2.1.2)

[error] Shouldn't redeclare 'person'. Consider to delete it or rename it.

'person' is defined here:


(lint/suspicious/noRedeclare)